.github/workflows/codegen.yml

```diff
@@ -0,0 +1,32 @@
+name: Codegen
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  codegen:
+    runs-on: ${{ matrix.os }}
+    name: Java ${{ matrix.java }} ${{ matrix.os }}
+    strategy:
+      matrix:
+        java: [17]
+        os: [macos-latest, ubuntu-latest, windows-latest]
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: gradle/wrapper-validation-action@v2
+
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v4
+        with:
+          java-version: ${{ matrix.java }}
+          distribution: 'corretto'
+
+      - name: Log Kotlin version
+        run: kotlin -version
+
+      - name: Clean and Build
+        run: ./gradlew clean build -Plog-tests
```

Comment:
Looks like the [noisy logs are still present in CI](https://github.com/smithy-lang/smithy-swift/actions/runs/8993406065/job/24705148277#step:6:23), though locally they are gone, e.g.

```log
> Task :buildSrc:compileKotlin
java.rmi.ServerError: Error occurred in server thread; nested exception is: 
Could not connect to kotlin daemon. Using fallback strategy.
	java.lang.AssertionError: symbolic reference class is not accessible: class sun.nio.ch.FileChannelImpl, from class org.jetbrains.kotlin.com.intellij.util.io.FileChannelUtil (unnamed module @46805db7)
	at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:389)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
```